open-company/open-company-web
# Performance improvement

Card: https://trello.com/c/vJY8xS8Z

Review with https://github.com/open-company/open-company-storage/compare/performance-improvement?expand=1

To test:
- open AP
- as soon as you see it click on the first post
- [ ] do you see the ghost posts but not the shaking carrot again?
- do you see 7 posts loaded per time?
- now use responsive mode from Chrome dev tools and make your window very tall that it fits at least the first 7 posts
- refresh the page
- [ ] do you see the app immediately loading the second batch of posts?
- [ ] can you scroll down and load more?
- add a reaction (should be immediate, not waiting for the network request
- add a comment (should be immediate, not waiting for the network request
- add a reaction to a comment (should be immediate, not waiting for the network request
- now open a post url
- now go to Network tab of Chrome dev tools and change the dropdown to 3G Slow
- add a comment
- [ ] do you see the new comment being added to the list and never disappearing (neither for a small amount of time)?
- remove 3g Slow
- force show the new version notification (in `oc.web.actions.web-app-update` remove 1000 from line 12 and 13, also change line 30 to `(if (not= status 404)`)
- load the app
- wait for the new version notification
- click it
- [ ] do you see the shaky carrot appear while it waits for the page refresh?
- [ ] does the click action seems more immediate than it used to be?
- now open a post with some comments
- turn off your network
- add a comment
- [ ] do you see the comment being reinserted in the comment field?
- [ ] do you still see the comments above the add comment field?

Regression test:
- open AP
- click in quick post
- add some body
- refresh page
- [ ] does it open back into edit mode with your changes visible?
- add some more text and some abstract and headline
- now refresh again from full post view
- [ ] does it open back into full post view with the latest changes?
